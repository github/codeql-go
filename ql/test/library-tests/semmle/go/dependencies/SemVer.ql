import semmle.go.dependencies.SemVer

from DependencySemVer ver, string normVer
where
  exists(int major, int minor, int patch |
    major = [0 .. 20] and minor = [0 .. 20] and patch = [0 .. 20]
  |
    normVer = major + "." + minor + "." + patch
  ) and
  ver.is(normVer)
select ver, normVer
