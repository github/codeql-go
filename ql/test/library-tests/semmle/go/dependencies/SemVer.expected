| v0.0.0-20150410010913-600d898af40a | 0.0.0 |
| v0.0.0-20150808065054-e02fc20de94c | 0.0.0 |
| v0.0.0-20160315200505-970db520ece7 | 0.0.0 |
| v0.0.0-20170225233418-6fe8760cad35 | 0.0.0 |
| v0.0.0-20170528113821-0c8571ac0ce1 | 0.0.0 |
| v0.0.0-20170604230408-02dd45c33376 | 0.0.0 |
| v0.0.0-20171010120322-cdade1c07385 | 0.0.0 |
| v0.0.0-20171227194143-aafc9e6bc7b7 | 0.0.0 |
| v0.0.0-20180219093839-391726f8973d | 0.0.0 |
| v0.0.0-20180306012644-bacd9c7ef1dd | 0.0.0 |
| v0.0.0-20180323000720-5d63dbd981b5 | 0.0.0 |
| v0.0.0-20180827131323-e3636079e1a4 | 0.0.0 |
| v0.0.0-20180830174826-7338bd80e790 | 0.0.0 |
| v0.0.0-20181003013248-f5e5bdd77824 | 0.0.0 |
| v0.0.0-20181026183834-f60e5f99f081 | 0.0.0 |
| v0.0.0-20181106222321-ec9c9a553398 | 0.0.0 |
| v0.0.0-20190103105442-ea782b38602d | 0.0.0 |
| v0.0.0-20190104093907-fbd5963f41e1 | 0.0.0 |
| v0.0.0-20190125204638-b490519efff9 | 0.0.0 |
| v0.0.0-20190216163058-9e08161f2597 | 0.0.0 |
| v0.0.0-20190624001437-99c81de45f40 | 0.0.0 |
| v0.0.0-20190707114632-bbf5a6c351f4 | 0.0.0 |
| v0.0.0-20200302225559-9b52d559c609 | 0.0.0 |
| v0.0.3 | 0.0.3 |
| v0.0.5 | 0.0.5 |
| v0.2.0 | 0.2.0 |
| v0.3.0 | 0.3.0 |
| v0.7.0 | 0.7.0 |
| v0.7.4 | 0.7.4 |
| v0.8.0 | 0.8.0 |
| v1.0.0 | 1.0.0 |
| v1.0.1 | 1.0.1 |
| v1.0.2 | 1.0.2 |
| v1.0.3 | 1.0.3 |
| v1.1.0 | 1.1.0 |
| v1.1.6 | 1.1.6 |
| v1.2.0 | 1.2.0 |
| v1.2.1 | 1.2.1 |
| v1.2.2 | 1.2.2 |
| v1.3.0 | 1.3.0 |
| v1.4.0 | 1.4.0 |
| v1.4.2 | 1.4.2 |
| v1.7.2 | 1.7.2 |
| v1.15.0 | 1.15.0 |
| v2.0.0+incompatible | 2.0.0 |
| v2.0.2+incompatible | 2.0.2 |
| v2.1.0+incompatible | 2.1.0 |
| v3.1.1+incompatible | 3.1.1 |
| v3.3.2+incompatible | 3.3.2 |
| v11.1.1+incompatible | 11.1.1 |
